Custom DC: Add BT automation and initial reference data to install script

Fructokinase opened the pull request:
Clone tools repo and call BT automation Terraform module. Also copy from the reference folder in datcom-public bucket into a folder called "reference" in custom dc resource bucket. 

Note that the reference bucket is currently empty and will be filled with actual sample tmcf shortly.

shifucun commented on scripts/install_custom_dc.sh:
```diff
@@ -83,6 +85,25 @@ terraform init && terraform apply \
+
+# TODO(alex): support custom robot SA and resource bucket name.
+WEBSITE_ROBOT="website-robot@$PROJECT_ID.google.com.iam.gserviceaccount.com"
+RESOURCE_BUCKET="$PROJECT_ID-resources"
```
do you need to create the bucket first?

Fructokinase replied:
Please see the comment below.

shifucun commented:
I tried the script.
1. There is a prompt to confirm, should skip that.
2. Got an error `Error: Error uploading object cloud_functions/bt_automation_go_source_chQTX3gEq8O6hb7ti7Qy3DJt6q3JRKJjmnx6AlPv4Qo=.zip: googleapi: Error 404: The specified bucket does not exist., notFound`

Fructokinase commented:
> I tried the script.
> 
> 1. There is a prompt to confirm, should skip that.
> 2. Got an error `Error: Error uploading object cloud_functions/bt_automation_go_source_chQTX3gEq8O6hb7ti7Qy3DJt6q3JRKJjmnx6AlPv4Qo=.zip: googleapi: Error 404: The specified bucket does not exist., notFound`

BT automation was originally going to be a part of setup. Seems like I removed the bucket when I moved Terraform module to tools repo. Added it back and updated the PR

shifucun commented:
Another error `googleapi: Error 403: Cloud Functions API has not been used in project 507955991373 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudfunctions.googleapis.com/overview?project=507955991373 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.`

shifucun commented on scripts/install_custom_dc.sh:
```diff
@@ -83,6 +85,26 @@ terraform init && terraform apply \
+git clone https://github.com/datacommonsorg/tools
+
+# TODO(alex): support custom robot SA and resource bucket name.
+WEBSITE_ROBOT="website-robot@$PROJECT_ID.google.com.iam.gserviceaccount.com"
```
should remove ".google.com" in WEBSITE_ROBOT

shifucun commented:
Another API issue `Error waiting for Creating CloudFunctions Function: Error code 7, message: Build failed: Cloud Build API has not been used in project 507955991373 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudbuild.googleapis.com/overview?project=507955991373 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.`

shifucun commented on scripts/install_custom_dc.sh:
```diff
@@ -83,6 +85,26 @@ terraform init && terraform apply \
+  -auto-approve
+
+# Copy over sample tmcfs/csvs from reference resource bucket.
+gsutil cp -r \
```
gsutil cp does not seem to work..

Fructokinase replied:
ack. If you mean entire folder copied over as opposed to files, folder levels are being copied over. As for the data, I will populate it shortly.

Fructokinase commented:
Added permissions for datcom SA and also enabled cloud functions API. As for the publisher, I believe that will be be needed by the identity of the script that manually triggers the controller, so no need to set it here?

shifucun commented:
> Added permissions for datcom SA and also enabled cloud functions API. As for the publisher, I believe that will be be needed by the identity of the script that manually triggers the controller, so no need to set it here?

Yes, sg.
